Issue 370 support reveal js 4 x (#430)
asciidoctor/asciidoctor-reveal.js: docs/modules/converter/pages/syntax/auto-animate.adoc

= Auto Animate

This feature is a wrapper of reveal.js auto animation.

> Since your adoc file will be transformed into HTML slide file, 

Reviewer: ```suggestion
Since your AsciiDoc file will be transformed into HTML slide file, 
```

> all settings will be mapped to `data-*` HTML attributes.

Reviewer: That's not completely true. I think we should talk in general terms like:

> You can define Asciidoctor attributes, options and IDs to configure the reveal.js auto-animate feature.

Author: Text revised.

> For bool attributes which means true if you set it, or false if you omit it,

Reviewer: I think it's misleading to talk about "attributes" since we don't know if we are talking about Asciidoctor attributes or reveal.js attributes.

I think we should talk about reveal.js settings and keep the term attributes when we are referring to Asciidoctor attributes.

Author: HTML property is used to distinguish from asciidoctor attributes.

you should use asciidoctor option, such as: `%auto-animate`.

> For attributes which need to set specific values, you can use asciidoctor attribute,
> such as: `auto-animate-easing='ease-in-out'`.

Reviewer: ```suggestion
For reveal.js settings which need to set specific values, you can use Asciidoctor attributes,
such as: `auto-animate-easing=ease-in-out`.
```

> == Enable animate

Reviewer: ```suggestion
== Enable auto-animate
```

> All you need to do is add %auto-animate to two adjacent slides.

Reviewer: ```suggestion
To enable the reveal.js auto-animate feature, you can add the `%auto-animate` option on adjacent elements.

In the following example, we are using the `%auto-animate` option on two adjacent sections:
```

----
[%auto-animate]
== auto-animate option

Matched element will be animated automatically!

[%auto-animate]
== auto-animate option

And move to the right position!

This is Great!

Matched element will be animated automatically!
----

== Control unmatched element animation

By default, no matching element will fade in.
You can make them appear instantly with: `auto-animate-unmatched=false`.

----
[%auto-animate,auto-animate-unmatched=false]
== auto-animate-unmatched

Determines whether elements with no matching auto-animate target should fade in. 
Set to false to make them appear instantly.

[%auto-animate,auto-animate-unmatched=false]
== auto-animate-unmatched

This will be shown instantly.

Determines whether elements with no matching auto-animate target should fade in. 
Set to false to make them appear instantly.
----

== Set animation duration

> You can set auto-animate-duration asciidocotr attribute to control the animation duration:

Reviewer: ```suggestion
You can set the `auto-animate-duration` attribute to control the animation duration:
```

----
[%auto-animate,auto-animate-duration=5]
== auto-animate-duration

Animation duration in seconds.

[%auto-animate,auto-animate-duration=5]
== auto-animate-duration

This auto animation 

will take 5 second.

Animation duration in seconds.
----

== Animation group

> You can separate groups of auto-animated slides, which will distinct from each other by `auto-animate-id` asciidocotor attribute.

Reviewer: ```suggestion
You can separate groups of auto-animated slides, which will distinct from each other by using the `auto-animate-id` attribute.
```

[%auto-animate,auto-animate-id="three"]
== auto-animate-id and %auto-animate-restart

`%auto-animate,auto-animate-id="three"`

If same Id, animation will work.

When you want separate groups of auto-animated slides right next to each other you can use the auto-animate-id and auto-animate-restart attributes.

[%auto-animate%auto-animate-restart,auto-animate-id="three"]
== auto-animate-id and %auto-animate-restart

`%auto-animate%auto-animate-restart,auto-animate-id="three"`

auto-animate-restart will prevent auto-animate between the previous slide (even if they have the same id) but not between it and the next slide.

When you want separate groups of auto-animated slides right next to each other you can use the auto-animate-id and auto-animate-restart attributes.
----

== Set animation easeing function

To use `auto-animate-easing` asciidoctor attribute, you can set
auto animation easing function.

----
[%auto-animate,auto-animate-easing='ease-in-out']
== auto-animate-easing

ease in out!

[%auto-animate,auto-animate-easing='ease-in-out']
== auto-animate-easing

This is GREAT!

The moving speed could

be controlled by CSS transition-timing-function

ease in out!
----

> 
> 
> 
> 
> 
> 
> 

Reviewer: ```suggestion
```
